File: backend/api/v1/ats.py
from fastapi import APIRouter

from backend.schemas.ats import ATSRequest
from backend.use_cases.analyze_ats_use_case import AnalyzeATSUseCase

router = APIRouter(
    prefix="/ats",
    tags=["ATS"],
)


@router.post("/analyze")
async def analyze_ats(
    request: ATSRequest,
):

    use_case = AnalyzeATSUseCase()

    result = use_case.execute(
        request.resume_data,
        request.job_description,
    )

    return result

File: web/src/services/ats.ts
import api from "./api";

export async function analyzeATS(
  resumeData: any,
  jobDescription = ""
) {
  const response = await api.post("/ats/analyze", {
    resume_data: resumeData,
    job_description: jobDescription,
  });

  return response.data;
}